src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,80 @@
+package lotto.controller;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import lotto.domain.*;
+import lotto.dto.LottoResultDto;
+import lotto.dto.TicketDto;
+import lotto.view.LottoView;
+
+public class LottoController {
+    private final LottoTicketSeller seller;
+    private final LottoView view;
+
+    public LottoController() {
+        this.seller = new LottoTicketSeller();
+        this.view = new LottoView();
+    }
+
+    public void start() {
+        LottoPurchaseBudget budget = getBudget();
+        List<LottoTicket> tickets = seller.generateTickets(budget);
+
+        view.printTickets(mapToTicketDto(tickets));
+
+        WinningLotto winningLotto = getWinningLotto();
+        LottoResult lottoResult = new LottoResult(winningLotto, tickets);
+
+        view.printLottoResult(mapToResultDto(lottoResult, budget));
+    }
+
+    private LottoPurchaseBudget getBudget() {
+        try {
+            return new LottoPurchaseBudget(view.getBudget());
+        } catch (RuntimeException e) {
+            view.printError(e);
+            return getBudget();
+        }
+    }
+
+    private WinningLotto getWinningLotto() {
+        try {
+            List<Integer> inputNumbers = view.getNumbers();
+            int bonusNumber = view.getBonusNumber();
+
+            List<LottoNumber> lottoNumbers = inputNumbers.stream()
+                    .map(LottoNumber::valueOf)
+                    .collect(Collectors.toList());
+
+            return new WinningLotto(lottoNumbers, LottoNumber.valueOf(bonusNumber));
+        } catch (RuntimeException e) {
+            view.printError(e);
+            return getWinningLotto();
+        }
+    }
+
+    private List<TicketDto> mapToTicketDto(List<LottoTicket> tickets) {
+        List<TicketDto> ticketDtos = new ArrayList<>();
+        for (LottoTicket ticket : tickets) {
+            List<Integer> numbers = ticket.toList().stream()
+                    .map(LottoNumber::getValue)
+                    .collect(Collectors.toList());
+
+            ticketDtos.add(new TicketDto(numbers));
+        }
+        return ticketDtos;
+    }
+
+    private LottoResultDto mapToResultDto(LottoResult lottoResult, LottoPurchaseBudget budget) {
+        Map<Prize, Long> result = lottoResult.getResult();
+
+        Map<String, Long> totalResult = new LinkedHashMap<>();
+        Arrays.stream(Prize.values())
+                .filter(prize -> prize != Prize.NOTHING)
+                .sorted((a, b) -> b.getOrder() - a.getOrder())
+                .forEach(prize -> totalResult.put(prize.toString(), result.getOrDefault(prize, 0L)));
```

Comment:
정렬 조건이 따로 없다면 아래와 같이 변경할 수 있을 것 같은데...
```suggestion
        Map<String, Long> totalResult = Arrays.stream(Prize.values())
            .filter(prize -> prize != Prize.NOTHING)
            .collect(Collectors.toMap(Prize::toString, prize -> result.getOrDefault(prize, 0L)));
```

정렬 조건을 유지한다면 아래와 같은 식으로 변경해볼 수 있을 것 같은데 오히려 조금 더 번거로워지는 느낌이라 차라리 지금의 방식을 유지하는게 조금 더 나아보이네요. 🙂
```suggestion
        Map<String, Long> totalResult = Arrays.stream(Prize.values())
            .filter(prize -> prize != Prize.NOTHING)
            .sorted((a, b) -> b.getOrder() - a.getOrder())
            .collect(Collectors.toMap(
                Prize::toString,
                prize -> result.getOrDefault(prize, 0L),
                (before, after) -> before,
                LinkedHashMap::new)
            );
```

또 한편으로는 생각해볼 부분은 View는 지금처럼 서버단과 동일한 환경(= SSR)일 수도 있고  프론트나 모바일에서 처리할 수도 있을 것 같은데, 보낼 때는 자료구조까지 생각해서 순서대로 보냈지만 수신하는 쪽의 환경에서는 다르게 받을 여지가 있는 것 같아요. 🤔

차라리 이런 경우라면 정렬한 결과를 List로 보내주면 수신하는 클라 측에서는 그대로 배열요소를 화면으로 뿌려주기만하면되니까 이런 문제가 적어지지 않을까 싶어요. 🤔